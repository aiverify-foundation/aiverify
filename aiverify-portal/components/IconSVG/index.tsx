import { ReactElement } from 'react';
import styles from './styles/icon.module.css';

import { AlertIcon } from './icons/alert-icon';
import { ArrowLeftIcon } from './icons/arrow-left-icon';
import { WideArrowLeftIcon } from './icons/wide-arrow-left-icon';
import { ArrowRightIcon } from './icons/arrow-right-icon';
import { WideArrowRightIcon } from './icons/wide-arrow-right-icon';
import { AsteriskIcon } from './icons/asterisk-icon';
import { BellIcon } from './icons/bell-icon';
import { BookIcon } from './icons/book-icon';
import { SolidBoxIcon } from './icons/solid-box-icon';
import { OutlineBoxIcon } from './icons/outline-box-icon';
import { BurgerMenuIcon } from './icons/burger-menu-icon';
import { ChatBubbleWideIcon } from './icons/chat-bubble-wide-icon';
import { ChatBubblesIcon } from './icons/chat-bubbles-icon';
import { CheckListIcon } from './icons/checklist-icon';
import { CircleArrowLeftIcon } from './icons/circle-arrow-left-icon';
import { CircleArrowRightIcon } from './icons/circle-arrow-right-icon';
import { CloseIcon } from './icons/close-x-icon';
import { DarkMoonIcon } from './icons/dark-moon-icon';
import { DocumentIcon } from './icons/document-icon';
import { FileIcon } from './icons/file-icon';
import { FolderForChatSessionsIcon } from './icons/folder-chat-icon';
import { FolderIcon } from './icons/folder-icon';
import { HistoryClockIcon } from './icons/history-clock-icon';
import { LayoutColumnsIcon } from './icons/layout-columns-icon';
import { LayoutWtfIcon } from './icons/layout-wtf-icon';
import { LightBulb } from './icons/light-bulb-icon';
import { LightSunIcon } from './icons/light-sun-icon';
import { LightningIcon } from './icons/lightning-icon';
import { ListIcon } from './icons/list-icon';
import { MaximizeIcon } from './icons/maximize-icon';
import { MinimizeIcon } from './icons/minimize-icon';
import { MoonshotAttackStrategyIcon } from './icons/ms-attack-icon';
import { MoonshotPromptTemplateIcon } from './icons/ms-prompt-template-icon';
import { PencilIcon } from './icons/pencil-icon';
import { PlusIcon } from './icons/plus-icon';
import { ResetIcon } from './icons/reset-icon';
import { RibbonIcon } from './icons/ribbon-icon';
import { RunCookbookIcon } from './icons/run-cookbook-icon';
import { SpacesuitIcon } from './icons/spacesuit-icon';
import { SquareIcon } from './icons/square-icon';
import { TableIcon } from './icons/table-icon';
import { TalkBubblesIcon } from './icons/talkbubbles-icon';
import { ToolsIcon } from './icons/tools-icon';
import { WarningIcon } from './icons/warning-icon';
import { WideArrowDownIcon } from './icons/wide-arrow-down';
import { WideArrowUpIcon } from './icons/wide-arrow-up';
import clsx from 'clsx';

enum IconName {
  Alert = 'Alert',
  ArrowLeft = 'ArrowLeft',
  ArrowRight = 'ArrowRight',
  Asterisk = 'Asterisk',
  Bell = 'Bell',
  Book = 'Book',
  BurgerMenu = 'BurgerMenu',
  ChatBubbleWide = 'ChatBubbleWide',
  ChatBubbles = 'ChatBubbles',
  CheckList = 'CheckList',
  CircleArrowLeft = 'CircleArrowLeft',
  CircleArrowRight = 'CircleArrowRight',
  Close = 'Close',
  DarkMoon = 'DarkMoon',
  Document = 'Document',
  File = 'File',
  Folder = 'Folder',
  FolderForChatSessions = 'FolderForChatSessions',
  HistoryClock = 'HistoryClock',
  LayoutColumns = 'LayoutColumns',
  LayoutWtf = 'LayoutWtf',
  LightBulb = 'LightBulb',
  LightSun = 'LightSun',
  Lightning = 'Lightning',
  List = 'List',
  Maximize = 'Maximize',
  Minimize = 'Minimize',
  MoonAttackStrategy = 'MoonAttackStrategy',
  MoonPromptTemplate = 'MoonPromptTemplate',
  OutlineBox = 'OutlineBox',
  Pencil = 'Pencil',
  Plus = 'Plus',
  Reset = 'Reset',
  Ribbon = 'Ribbon',
  RunCookbook = 'RunCookbook',
  SolidBox = 'SolidBox',
  Spacesuit = 'Spacesuit',
  Square = 'Square',
  Table = 'Table',
  TalkBubbles = 'TalkBubbles',
  Tools = 'Tools',
  Warning = 'Warning',
  WideArrowDown = 'WideArrowDown',
  WideArrowLeft = 'WideArrowLeft',
  WideArrowRight = 'WideArrowRight',
  WideArrowUp = 'WideArrowUp',
}

type BaseIconProps = {
  name: IconName;
  role?: string;
  ariaLabel?: string;
  size?: number;
  disabled?: boolean;
  style?: React.CSSProperties;
  onClick?: (e: React.MouseEvent<HTMLDivElement>) => void;
  onMouseDown?: (e: React.MouseEvent) => void;
};

type WithColor = BaseIconProps & {
  color: string;
  svgClassName?: never;
};

type WithClassName = BaseIconProps & {
  svgClassName: `stroke-${string} dark:stroke-${string}` | `fill-${string} dark:fill-${string}`;
  color?: never;
};

type IconProps = WithColor | WithClassName;

function Icon(props: IconProps) {
  const {
    name,
    role,
    ariaLabel,
    onClick,
    onMouseDown,
    disabled = false,
    size = 20,
    style,
    color = '#000000',
    svgClassName,
  } = props;

  let iconToRender: ReactElement | null = null;

  switch (name) {
    case IconName.Alert:
      iconToRender = (
        <AlertIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Folder:
      iconToRender = (
        <FolderIcon
          backColor="#2980b9"
          frontColor="#3498db"
          midColor="#bdc3c7"
          width={size}
          height={size}
        />
      );
      break;
    case IconName.FolderForChatSessions:
      iconToRender = (
        <FolderForChatSessionsIcon
          backColor="#2980b9"
          frontColor="#3498db"
          midColor="#bdc3c7"
          chatIconColor="#2980b9"
          width={size}
          height={size}
        />
      );
      break;
    case IconName.ChatBubbles:
      iconToRender = (
        <ChatBubblesIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.RunCookbook:
      iconToRender = (
        <RunCookbookIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.BurgerMenu:
      iconToRender = (
        <BurgerMenuIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.DarkMoon:
      iconToRender = (
        <DarkMoonIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.LightSun:
      iconToRender = (
        <LightSunIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Close:
      iconToRender = (
        <CloseIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.WideArrowLeft:
      iconToRender = (
        <WideArrowLeftIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.WideArrowRight:
      iconToRender = (
        <WideArrowRightIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.CircleArrowRight:
      iconToRender = (
        <CircleArrowRightIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.CircleArrowLeft:
      iconToRender = (
        <CircleArrowLeftIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Maximize:
      iconToRender = (
        <MaximizeIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Minimize:
      iconToRender = (
        <MinimizeIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.LayoutWtf:
      iconToRender = (
        <LayoutWtfIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.LayoutColumns:
      iconToRender = (
        <LayoutColumnsIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.ChatBubbleWide:
      iconToRender = (
        <ChatBubbleWideIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Reset:
      iconToRender = (
        <ResetIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.SolidBox:
      iconToRender = (
        <SolidBoxIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.OutlineBox:
      iconToRender = (
        <OutlineBoxIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Plus:
      iconToRender = <PlusIcon width={size} height={size} className={svgClassName} color={color} />;
      break;
    case IconName.Square:
      iconToRender = (
        <SquareIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.List:
      iconToRender = <ListIcon width={size} height={size} className={svgClassName} color={color} />;
      break;
    case IconName.File:
      iconToRender = <FileIcon width={size} height={size} className={svgClassName} color={color} />;
      break;
    case IconName.Book:
      iconToRender = <BookIcon width={size} height={size} className={svgClassName} color={color} />;
      break;
    case IconName.ArrowLeft:
      iconToRender = (
        <ArrowLeftIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.ArrowRight:
      iconToRender = (
        <ArrowRightIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Table:
      iconToRender = (
        <TableIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Bell:
      iconToRender = <BellIcon width={size} height={size} className={svgClassName} color={color} />;
      break;
    case IconName.Asterisk:
      iconToRender = (
        <AsteriskIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.CheckList:
      iconToRender = (
        <CheckListIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Spacesuit:
      iconToRender = (
        <SpacesuitIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.HistoryClock:
      iconToRender = (
        <HistoryClockIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Tools:
      iconToRender = (
        <ToolsIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Lightning:
      iconToRender = (
        <LightningIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.WideArrowDown:
      iconToRender = (
        <WideArrowDownIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.WideArrowUp:
      iconToRender = (
        <WideArrowUpIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.LightBulb:
      iconToRender = (
        <LightBulb width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Ribbon:
      iconToRender = (
        <RibbonIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.TalkBubbles:
      iconToRender = (
        <TalkBubblesIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Document:
      iconToRender = (
        <DocumentIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Warning:
      iconToRender = (
        <WarningIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Pencil:
      iconToRender = (
        <PencilIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.MoonPromptTemplate:
      iconToRender = (
        <MoonshotPromptTemplateIcon
          width={size}
          height={size}
          className={svgClassName}
          color={color}
        />
      );
      break;
    case IconName.MoonAttackStrategy:
      iconToRender = (
        <MoonshotAttackStrategyIcon
          width={size}
          height={size}
          className={svgClassName}
          color={color}
        />
      );
      break;
    default:
      iconToRender = null;
  }

  const cssClass = clsx(
    styles.icon_wrapper,
    onClick ? styles.pointer_effect : '',
    disabled ? styles.disabled : '',
  );

  return (
    <div
      role={role}
      aria-label={ariaLabel}
      className={cssClass}
      style={{
        width: size,
        height: size,
        ...style,
      }}
      onClick={disabled ? undefined : onClick}
      onMouseDown={disabled ? undefined : onMouseDown}
    >
      {iconToRender}
    </div>
  );
}

export { Icon, IconName };
