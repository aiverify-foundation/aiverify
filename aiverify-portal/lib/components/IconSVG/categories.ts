import { IconName } from '.';

export const lineIcons = [
  IconName.Alert,
  IconName.ArrowRight,
  IconName.ArrowLeft,
  IconName.Pencil,
  IconName.Asterisk,
  IconName.Book,
  IconName.BurgerMenu,
  IconName.ChatBubbleWide,
  IconName.ChatBubbles,
  IconName.CircleArrowRight,
  IconName.CircleArrowLeft,
  IconName.Close,
  IconName.Document,
  IconName.File,
  IconName.LayoutColumns,
  IconName.LayoutWtf,
  IconName.LightSun,
  IconName.List,
  IconName.Maximize,
  IconName.Minimize,
  IconName.OutlineBox,
  IconName.Pencil,
  IconName.Plus,
  IconName.Reset,
  IconName.RunCookbook,
  IconName.Square,
  IconName.Table,
  IconName.TalkBubbles,
  IconName.Warning,
  IconName.WideArrowUp,
  IconName.WideArrowDown,
  IconName.WideArrowLeft,
  IconName.WideArrowRight,
];

export const fillIcons = [
  IconName.Bell,
  IconName.CheckList,
  IconName.DarkMoon,
  IconName.Database,
  IconName.HistoryClock,
  IconName.LightBulb,
  IconName.Lightning,
  IconName.MoonPromptTemplate,
  IconName.Ribbon,
  IconName.SolidBox,
  IconName.Spacesuit,
  IconName.Tools,
  IconName.OpenedBook,
  IconName.OpenedEmptyBook,
  IconName.Plug,
];

export const colorIcons = [IconName.Folder, IconName.FolderForChatSessions];
