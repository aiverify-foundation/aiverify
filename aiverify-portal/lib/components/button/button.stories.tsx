import type { Meta, StoryObj } from '@storybook/react';
import { Button, ButtonVariant } from '.';
import { IconName } from '../IconSVG';

const meta: Meta<typeof Button> = {
  title: 'Components/Button',
  component: Button,
  parameters: {
    backgrounds: {
      default: 'transparent',
      values: [{ name: 'black', value: '#000000' }],
    },
    controls: {
      sort: 'requiredFirst',
      expanded: false,
    },
  },
  args: {
    text: 'Your Button',
    size: 'md',
    pill: false,
    bezel: true,
    disabled: false,
  },
  argTypes: {
    variant: {
      control: 'radio',
      options: Object.values(ButtonVariant).filter((variant) => variant !== ButtonVariant.OUTLINE),
    },
    size: {
      control: 'radio',
      options: ['lg', 'md', 'sm', 'xs'],
    },
    pill: {
      control: 'boolean',
    },
    icon: {
      control: 'select',
      options: Object.values(IconName),
    },
    width: {
      control: 'number',
      table: {
        disable: true,
      },
    },
    disabled: {
      control: 'boolean',
    },
  },
};

export default meta;
type Story = StoryObj<typeof Button>;

export const Bezel: Story = {
  args: {
    variant: ButtonVariant.PRIMARY,
  },
  argTypes: {
    text: {
      description: 'The text of the button',
    },
    variant: {
      options: [ButtonVariant.PRIMARY, ButtonVariant.SECONDARY],
      description: 'There are 5 variants built in.',
      table: {
        type: {
          summary: 'primary | secondary | outline | link | text',
        },
        defaultValue: {
          summary: 'primary',
        },
      },
    },
    size: {
      description: 'The size of the button. There are 4 sizes built in.',
      table: {
        type: {
          summary: 'lg | md | sm | xs',
        },
        defaultValue: {
          summary: 'md',
        },
      },
    },
    pill: {
      description: 'For a pill-shaped button',
      table: {
        defaultValue: {
          summary: 'false',
        },
      },
    },
    bezel: {
      description: 'Whether the button has a bezel or it is just flat',
      table: {
        disable: true,
      },
    },
    color: {
      table: {
        disable: true,
      },
    },
    hoverColor: {
      table: {
        disable: true,
      },
    },
    pressedColor: {
      table: {
        disable: true,
      },
    },
    textColor: {
      table: {
        disable: true,
      },
    },
    icon: {
      table: {
        disable: true,
      },
    },
    iconPosition: {
      table: {
        disable: true,
      },
    },
    iconSize: {
      table: {
        disable: true,
      },
    },
    iconColor: {
      table: {
        disable: true,
      },
    },
  },
};

export const Flat: Story = {
  args: {
    variant: ButtonVariant.PRIMARY,
    bezel: false,
  },
  argTypes: {
    variant: {
      options: [ButtonVariant.PRIMARY, ButtonVariant.SECONDARY, ButtonVariant.OUTLINE],
    },
    color: {
      table: {
        disable: true,
      },
    },
    hoverColor: {
      table: {
        disable: true,
      },
    },
    pressedColor: {
      table: {
        disable: true,
      },
    },
    textColor: {
      table: {
        disable: true,
      },
    },
    icon: {
      table: {
        disable: true,
      },
    },
    iconPosition: {
      table: {
        disable: true,
      },
    },
    iconSize: {
      table: {
        disable: true,
      },
    },
    iconColor: {
      table: {
        disable: true,
      },
    },
  },
};

export const WithIcon: Story = {
  args: {
    variant: ButtonVariant.PRIMARY,
    text: 'With Icon',
    iconPosition: 'left',
    icon: IconName.Close,
    iconSize: 18,
  },
  argTypes: {
    icon: {
      control: 'select',
      options: Object.values(IconName),
    },
    variant: {
      control: 'radio',
      options: Object.values(ButtonVariant),
    },
    color: {
      table: {
        disable: true,
      },
    },
    hoverColor: {
      table: {
        disable: true,
      },
    },
    pressedColor: {
      table: {
        disable: true,
      },
    },
    textColor: {
      table: {
        disable: true,
      },
    },
  },
};

export const Outline: Story = {
  args: {
    variant: ButtonVariant.OUTLINE,
    text: 'Your Button',
  },
  argTypes: {
    variant: {
      control: 'radio',
      options: [ButtonVariant.OUTLINE],
    },
    bezel: {
      table: {
        disable: true,
      },
    },
    icon: {
      table: {
        disable: true,
      },
    },
    iconPosition: {
      table: {
        disable: true,
      },
    },
    iconSize: {
      table: {
        disable: true,
      },
    },
    color: {
      table: {
        disable: true,
      },
    },
    hoverColor: {
      table: {
        disable: true,
      },
    },
    pressedColor: {
      table: {
        disable: true,
      },
    },
    textColor: {
      table: {
        disable: true,
      },
    },
  },
};

export const Link: Story = {
  parameters: {
    backgrounds: {
      default: 'white',
      values: [{ name: 'black', value: '#000000' }],
    },
    controls: {
      sort: 'requiredFirst',
      expanded: false,
    },
  },
  args: {
    variant: ButtonVariant.LINK,
    text: 'This is a link button',
  },
  argTypes: {
    text: {
      description: 'The text of the button',
    },
    variant: {
      description: 'There are 5 variants built in.',
      control: 'radio',
      options: [ButtonVariant.LINK],
      table: {
        type: {
          summary: 'primary | secondary | outline | link | text',
        },
        defaultValue: {
          summary: 'primary',
        },
      },
    },
    size: {
      description: 'The size of the button. There are 4 sizes built in.',
      table: {
        type: {
          summary: 'lg | md | sm | xs',
        },
        defaultValue: {
          summary: 'md',
        },
      },
    },
    pill: {
      description: 'For a pill-shaped button',
      table: {
        defaultValue: {
          summary: 'false',
        },
      },
    },
    bezel: {
      description: 'Whether the button has a bezel or it is just flat',
      table: {
        disable: true,
      },
    },
    color: {
      table: {
        disable: true,
      },
    },
    hoverColor: {
      table: {
        disable: true,
      },
    },
    pressedColor: {
      table: {
        disable: true,
      },
    },
    textColor: {
      table: {
        disable: true,
      },
    },
    icon: {
      table: {
        disable: true,
      },
    },
    iconPosition: {
      table: {
        disable: true,
      },
    },
    iconSize: {
      table: {
        disable: true,
      },
    },
    iconColor: {
      table: {
        disable: true,
      },
    },
  },
};

export const Customized: Story = {
  args: {
    variant: ButtonVariant.PRIMARY,
    text: 'Customized Pill Shaped',
    iconPosition: 'left',
    icon: IconName.Close,
    color: '#524e56',
    hoverColor: '#464349',
    pressedColor: '#3d3a40',
    textColor: '#ffffff',
    pill: true,
    bezel: false,
    width: 280,
    iconSize: 22,
  },
  argTypes: {
    width: {
      control: 'number',
      table: {
        disable: false,
      },
    },
  },
};