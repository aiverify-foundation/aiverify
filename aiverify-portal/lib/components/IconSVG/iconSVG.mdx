import { Meta, Title, Primary, Stories, Controls } from '@storybook/blocks';
import IconSVGStories from './iconSVG.stories';
import { Icon, IconName } from './index';
import { lineIcons, fillIcons, colorIcons } from './categories';


<Title>SVG Icons React Component</Title>
<Meta of={IconSVGStories} />
<Primary />
<Controls />
## Icon Names

<div style={{
  marginTop: '2rem',
  display: 'grid',
  gridTemplateColumns: 'repeat(8, 1fr)',
  width: '100%',
  rowGap: '2rem',
}}>
  {lineIcons.map((iconName) => (
    <div key={iconName} style={{ display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
      <Icon name={iconName} size={50} color="#000000" />
      <span style={{ fontSize: '0.8rem', textAlign: 'center', marginTop: '0.5rem' }}>{iconName}</span>
    </div>
  ))}
  {fillIcons.map((iconName) => (
    <div key={iconName} style={{ display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
      <Icon name={iconName} size={50} color="#000000" />
      <span style={{ fontSize: '0.8rem', textAlign: 'center', marginTop: '0.5rem' }}>{iconName}</span>
    </div>
  ))}
  {colorIcons.map((iconName) => (
    <div key={iconName} style={{ display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
      <Icon name={iconName} size={50} color="#000000" />
      <span style={{ fontSize: '0.8rem', textAlign: 'center', marginTop: '0.5rem' }}>{iconName}</span>
    </div>
  ))}

</div>
