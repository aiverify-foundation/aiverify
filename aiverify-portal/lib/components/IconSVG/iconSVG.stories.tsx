import type { Meta, StoryObj } from '@storybook/react';
import { Icon, IconName } from '.';
import { fillIcons, lineIcons } from './categories';

const meta: Meta<typeof Icon> = {
  title: 'Components/SVG Icons',
  component: Icon,
  parameters: {
    controls: {
      sort: 'requiredFirst',
      expanded: true,
    },
  },
  argTypes: {
    name: {
      control: 'select',
      options: Object.values(IconName).sort(),
      description: 'Name of the icon',
    },
    size: {
      control: 'number',
      description: 'Size of the icon in CSS px',
    },
    svgClassName: {
      control: 'text',
      description: 'Class name for the SVG',
      table: {
        type: { summary: 'string' },
        disable: true,
      },
    },
    role: {
      table: {
        disable: true,
      },
    },
    ariaLabel: {
      table: {
        disable: true,
      },
    },
    style: {
      table: {
        disable: true,
      },
    },
    onMouseDown: {
      table: {
        disable: true,
      },
    },
    onClick: {
      table: {
        disable: true,
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Icon>;

export const Line_Icon: Story = {
  argTypes: {
    name: {
      control: 'select',
      options: lineIcons.sort(),
      description: 'Name of the icon',
    },
  },
  args: {
    name: IconName.Alert,
    size: 50,
    svgClassName: 'stroke-primary-900 dark:stroke-primary-50',
    onClick: () => alert('Icon clicked'),
  },
};

export const Fill_Icon: Story = {
  argTypes: {
    name: {
      control: 'select',
      options: fillIcons.sort(),
      description: 'Name of the icon',
    },
  },
  args: {
    name: IconName.Bell,
    size: 50,
    svgClassName: 'fill-primary-900 dark:fill-primary-50',
    onClick: () => alert('Icon clicked'),
  },
};

export const Color_Icon: Story = {
  args: {
    name: IconName.Folder,
    size: 50,
    svgClassName: 'fill-primary-900 dark:fill-primary-50',
    onClick: () => alert('Icon clicked'),
  },
  argTypes: {
    name: {
      control: 'select',
      options: [IconName.Folder, IconName.FolderForChatSessions],
      description: 'Name of the icon',
    },
  },
};
