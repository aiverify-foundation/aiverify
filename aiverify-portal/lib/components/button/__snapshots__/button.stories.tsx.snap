// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Button Bezel smoke-test 1`] = `
<div style="position: relative; width: 100%; height: 300px; display: flex; align-items: center; justify-content: center;">
  <button type="button"
          class="_btn_1vb4e_1 _btn_primary_1vb4e_35 _btn_md_1vb4e_199 _bezel_1vb4e_35"
  >
    Your Button
  </button>
</div>
`;

exports[`Components/Button Customized smoke-test 1`] = `
<div style="position: relative; width: 100%; height: 300px; display: flex; align-items: center; justify-content: center;">
  <button type="button"
          class="_btn_1vb4e_1 _btn_primary_1vb4e_35 _btn_md_1vb4e_199 _pill_1vb4e_259 _flat_1vb4e_87"
          style="background: rgb(82, 78, 86); color: rgb(255, 255, 255); width: 280px;"
  >
    <div class="_icon_wrapper_1j07s_1"
         style="width: 22px; height: 22px;"
    >
      <svg width="22"
           height="22"
           viewbox="0 0 32 32"
           stroke="#ffffff"
           fill="none"
      >
        <g id="cross">
          <line stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                x1="7"
                x2="25"
                y1="7"
                y2="25"
          >
          </line>
          <line stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                x1="7"
                x2="25"
                y1="25"
                y2="7"
          >
          </line>
        </g>
      </svg>
    </div>
    Customized Pill Shaped
  </button>
</div>
`;

exports[`Components/Button Flat smoke-test 1`] = `
<div style="position: relative; width: 100%; height: 300px; display: flex; align-items: center; justify-content: center;">
  <button type="button"
          class="_btn_1vb4e_1 _btn_primary_1vb4e_35 _btn_md_1vb4e_199 _flat_1vb4e_87"
  >
    Your Button
  </button>
</div>
`;

exports[`Components/Button Link smoke-test 1`] = `
<div style="position: relative; width: 100%; height: 300px; display: flex; align-items: center; justify-content: center;">
  <button type="button"
          class="_btn_1vb4e_1 _btn_link_1vb4e_283 _btn_md_1vb4e_199 _bezel_1vb4e_35"
  >
    This is a link button
  </button>
</div>
`;

exports[`Components/Button Outline smoke-test 1`] = `
<div style="position: relative; width: 100%; height: 300px; display: flex; align-items: center; justify-content: center;">
  <button type="button"
          class="_btn_1vb4e_1 _btn_outline_1vb4e_139 _btn_md_1vb4e_199 _flat_1vb4e_87"
  >
    Your Button
  </button>
</div>
`;

exports[`Components/Button WithIcon smoke-test 1`] = `
<div style="position: relative; width: 100%; height: 300px; display: flex; align-items: center; justify-content: center;">
  <button type="button"
          class="_btn_1vb4e_1 _btn_primary_1vb4e_35 _btn_md_1vb4e_199 _bezel_1vb4e_35"
  >
    <div class="_icon_wrapper_1j07s_1"
         style="width: 18px; height: 18px;"
    >
      <svg width="18"
           height="18"
           viewbox="0 0 32 32"
           stroke="#ffffff"
           fill="none"
      >
        <g id="cross">
          <line stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                x1="7"
                x2="25"
                y1="7"
                y2="25"
          >
          </line>
          <line stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                x1="7"
                x2="25"
                y1="25"
                y2="7"
          >
          </line>
        </g>
      </svg>
    </div>
    With Icon
  </button>
</div>
`;
